source/libidb/libidb.cpp

```diff
@@ -99,7 +99,6 @@ size_t _BDATA::grow( size_t new_real )
 		if( data_buff != NULL )
 		{
 			memcpy( new_buff, data_buff, data_real );
-			delete [] data_buff;

```

Comment:
The `double free or corruption` error occurs when memory is freed twice or when memory is freed that was not previously allocated. It seems that the `data_buff` pointer is being freed twice: once in the `grow` method and again in the `~_BDATA` destructor.

To fix that we can just don't try to free the memory in the `grow` method.

Reply:
I think your changes would hide an already-existing use-after-free bug. If "data_buff" was already freed, and we just copied data out of it, then what the heck did we just copy?

I'm working on my own fix right now. The root cause in my test case (missing configuration data) appears to be in _CONFIG_MANAGER::file_vpn_load. When config.get_ispublic() is true, if sites_all isn't set, then it adds an empty string to the path. That alone isn't a problem, but then it goes to try to add a delimiter to "size-1", which causes an integer overflow. It doesn't appear ins was written to detect this edge case.